sootup.java.bytecode/src/main/java/sootup/java/bytecode/frontend/AsmUtil.java

```diff
@@ -154,60 +154,96 @@ public static Type toJimpleSignature(@Nonnull String desc) {
 
   @Nonnull
   public static Type toJimpleType(@Nonnull String desc) {
-    int idx = desc.lastIndexOf('[');
-    int nrDims = idx + 1;
+    int nrDims = countArrayDim(desc);
     if (nrDims > 0) {
-      if (desc.charAt(0) != '[') {
+      desc = desc.substring(nrDims);
+    }
+
+    Type baseType = toPrimitiveOrVoidType(desc).orElse(null);
+    if (baseType == null && desc.charAt(0) != 'L') {
+      throw new AssertionError("Unknown descriptor: " + desc);
+    }
+
+    if (baseType == null && desc.charAt(0) == 'L') {
```

Comment:
please merge the same checks on: baseType==null

Reply:
so the second half becomes the else part as well

Reply:
Nope there are 4 different case :) 
But yeah I can put them into a if (baseType==null)....